cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/resource/volume/AwsVolumeResourceBuilder.java

```diff
@@ -238,7 +238,7 @@ private String getAvailabilityZoneFromSubnet(AuthenticatedContext auth, String s
         }
         LOGGER.debug("Volume creation requests sent");
 
-        return buildableResource.stream()
+        return requestedResources.stream()
```

Comment:
I am not sure if this is correct. During repair, the context is populated with the already existing disks. I think the status of those is not REQUESTED, rather DETACHED or stg like that. During repair, those are just passed through this builder without modification and the attachment builder will be responsible to attach those disks to an instance. Have you tried repair with this change?

Reply:
I'm implementing the repair with multi-az.
I have found a bug if any errors occur during the repair flow, the DETACHED volumes statuses will be set to CREATED. After the restart, the resource builder won't find the disk because we try to find it with DETACHED status and the volumes won't reattach but create attached disks again.

Reply:
It worked until an error because we fetched at the beginning of the flow step and we worked with the "outdated" entity. I mean we only updated in the DB the status of the resource but in the memory, we used it with the old status